lua/entities/gmod_wire_egp/lib/egplib/usefulfunctions.lua

```diff
@@ -155,12 +155,69 @@ function EGP:SendMaterial( obj ) -- ALWAYS use this when sending material
 	end
 end
 
+EGP.CachedMaterials = {}
+EGP.LoadingMaterials = {}
+
+function EGP:CheckURLDownloads()
+	for mat, v in pairs(EGP.LoadingMaterials) do
+		if not v.Panel:IsLoading() then
+			timer.Simple(0.2,function()
+				local ShaderInfo = {
+				["$vertexcolor"] = 1,
+				["$vertexalpha"] = 1,
+				["$ignorez"] = 1,
+				["$nolod"] = 1,
+				}
+				local urlmaterial = CreateMaterial("egp_urltex_" .. util.CRC(mat .. SysTime()), "UnlitGeneric", ShaderInfo)
+				local tex = v.Panel:GetHTMLMaterial():GetTexture("$basetexture")
+				tex:Download()
+				urlmaterial:SetTexture("$basetexture", tex)
+				tex:Download()
```

Comment:
I don't know. I found that snippet in CapsAdmin's Pac3 mod. I was also confused and I can't find any documentation on it.